program-runtime/src/invoke_context.rs

```diff
@@ -591,7 +593,12 @@ impl<'a> InvokeContext<'a> {
 
         // Verify the per-account instruction results
         let (mut pre_sum, mut post_sum) = (0_u128, 0_u128);
-        let mut work = |_index_in_instruction: usize, instruction_account: &InstructionAccount| {
+        'outer_loop: for (index_in_instruction, instruction_account) in
+            instruction_accounts.iter().enumerate()
+        {
+            if index_in_instruction != instruction_account.index_in_callee {
```

Comment:
Copy pasta and still be very clear what's going on here

Reply:
I see 4 ways to make it easier to understand:
- Add a comment, e.g. `continue; // Skip duplicate account`
- Replace the first continue by `.filter()` and the rest of the loop by `.map()`
- Refactor it back into a function, which would bring back the problems with the `Result` return type
- Use a macro to avoid having a lambda and the issues with the `Result` return type

Reply:
Yes, at least comment

with a comment is probably ok, would be interesting to see the macro